ape_safe/accounts.py

```diff
@@ -276,11 +273,27 @@ def client(self) -> BaseSafeClient:
         return SafeClient(address=self.address, chain_id=self.provider.chain_id)
 
     @property
-    def version(self) -> str:
-        try:
-            return self.client.safe_details.version.replace("+L2", "")
-        except Exception:
-            return self.contract.VERSION()
+    def version(self) -> Version:
+        # NOTE: We want to make a direct call, so we can use this for loading
+        #       the proper verison of the Safe protocol w/ `.contract`
+        VERSION_ABI = MethodABI(
+            name="VERSION",
+            type="function",
+            stateMutability="view",
+            outputs=[ABIType(type="string")],
+        )
+
+        if isinstance(version := ContractCall(VERSION_ABI, address=self.address)(), str):
+            return Version(version)
+
+        # NOTE: If `eth_call` returns nothing, it will be rendered as randomly
+        raise ContractNotFoundError(
+            self.address,
+            bool(self.provider.network.explorer),
+            f"{self.provider.network.ecosystem.name}:"
```

Comment:
I think you can just do `self.provider.network_choice` here

Reply:
just discovered this one the other day

Reply:
df86de5